content/docs/querying/operators.md

```diff
@@ -71,10 +76,63 @@ over from the left-hand-side vector.
 
 `vector1 or vector2` results in a vector that contains all original elements
 (labelsets + values) of `vector1` and additionally all elements of `vector2`,
-which don't have matching labelsets in `vector1`. The metric name is carried
-over from the left-hand-side vector in both cases.
+which don't have matching labelsets in `vector1`.
+
+## Vector matching
+
+Operations between vectors attempt to find a matching element in the right-hand-side
+vector for each entry in the left-hand-side. There are two basic types of
+matching behaviour:
+
+**One-to-one** finds a unique pair of entries from each side of the operation.
+In the default case, that is an operation following the format `vector1 <operator> vector2`.
+Two entries match if they have the exact same set of labels and corresponding values.
+By means of the `on` keyword one can reduce the set of considered labels to a provided list:
+
+  <vector expr> <bin-op> on(<label list>) <vector expr>
+
+_Modified matching methods can only be used for [comparison/filterting](#comparison-/-filter-binary-operators)
+and [airthmetic](#arithmetic-binary-operators) operations._
+
+Example:
+
+  http_errors_total / on(method) http_requests_total
```

Comment:
Any examples should follow best practice, so this example needs a sumb by(method)(rate()) on both sides